[Feat] Add getMapboxRef prop

robert-claypool opened the pull request:
Adds a `getMapboxRef` prop which you can set to your custom function. That function is passed down into KeplerGl's MapContainer and is used as a callback to expose the MapContainer's `react-map-gl` [InteractiveMap](https://uber.github.io/react-map-gl/#/Documentation/api-reference/interactive-map) instance. In other words, `getMapboxRef` gives KeplerGl's parent component a reference to the inner Mapbox map.

Is this a good pattern or an anti-pattern? See my update to `examples/demo-app/src/app.js` on how it would be used.

See #268. Exposing access to the full [Mapbox GL API](https://docs.mapbox.com/mapbox-gl-js/api/) should also open a solution for #293 and #118.

Kepler.gl has powerful client-side visualizations for GeoJSON. By having access to the Mapbox GL API, developers can easily add [raster](https://docs.mapbox.com/mapbox-gl-js/example/map-tiles/) and vector layers from a variety of sources, and further customize map interactions.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/uber/kepler.gl?pullRequest=372) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/uber/kepler.gl?pullRequest=372) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/uber/kepler.gl?pullRequest=372) it.</sub>

robert-claypool commented:
I believe the build failure is unrelated to my diff.

robert-claypool commented:
@heshan0131, thank you for the review.

I believe 5975a95f6 addresses your concerns... from the extended commit message:
> This simplifies the handling of getMapboxRef so that KeplerGL forwards your callback function to MapContainer unchanged. It also adds support for multiple maps. Use 'index' (0 or 1) to identify which map triggered the callback. If 'mapbox' is null, then that map has been removed by toggleSplitMap.

robert-claypool commented:
README updated

heshan0131 commented:
Thanks, this looks good to me. I am working with the FOSSA team to fix the build, but we can land this now if needed

robert-claypool commented:
No hurry, I'll use my fork until version v0.2.3 lands.

robert-claypool commented:
The force-push was to rebase `robert-claypool/expose-mapbox-map` against `kepler.gl/master`.
